pkg/corerp/backend/controller/createorupdateresource.go

```diff
@@ -125,6 +127,8 @@ func (c *CreateOrUpdateResource) Run(ctx context.Context, request *ctrl.Request)
 		}
 	}
```

Comment:
If the deployment fails, we will still go ahead and delete the old versions of the working applications. See the line 123 where we do garbage collection above. I guess this is not the behavior we would like to have. We would like to keep the working application until we have a newer version that is working.